### ISSUE TITLE
SOQL does not validate when query from GROUP or ORDER objects

### ISSUE THREAD

**Issue by cchrisv** [douglascayers-org/sfdx-mass-action-scheduler#89]
I am getting the following error "**error Invalid SOQL Query Expected [0-9a-zA-Z_] but " " found.** " when attempting to run the following SOQL query "**SELECT DeveloperName FROM Group WHERE Type = 'Queue' AND QueueRoutingConfigId != null ORDER BY CreatedDate ASC** " The query runs fine in Developer Console.

**Comment by douglascayers** [comment]
Hi @cchrisv,

Thanks for your interest in Mass Action Scheduler. To help me troubleshoot, please answer the questions in the bug issue template found at https://github.com/douglascayers-org/sfdx-mass-action-scheduler/issues/new?assignees=douglascayers&labels=discussion+%F0%9F%92%AC&template=BUG_REPORT.md&title=

**Comment by douglascayers** [comment]
I think this may be a bug with the SOQL validator library I use, [soql-parse](https://github.com/stomita/soql-parse). I think it's seeing the object name `Group` and expecting it is trying to define the **GROUP BY** clause of the query.

This query works fine:
```
SELECT Name FROM Account WHERE Type = 'Queue' ORDER BY CreatedDate ASC
```

However, change the **FROM** object from `Account` to `Group` and I get your error:
```
SELECT Name FROM Group WHERE Type = 'Queue' ORDER BY CreatedDate ASC
```

**Comment by cchrisv** [comment]
Thanks for looking into this. I had it on my "to-do" list to fill out the full bug report, sorry.

**Comment by douglascayers** [comment]
I've submitted a PR to the soql-parse repo

**Comment by douglascayers** [comment]
Release 2.3 is now available per #99, please see the [release notes](https://github.com/douglascayers-org/sfdx-mass-action-scheduler/wiki/Release-Notes-v2.3) for full details. Thank you!
